fieldservice_stock/models/stock.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (C) 2018 - TODAY, Brian McMaster
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    fsm_order_line_id = fields.Many2one('fsm.order.line', 'FSM Order Line')
+    fsm_order_return_line_id = fields.Many2one(
+        'fsm.order.return', 'FSM Order Return Line')
+
+    def prepare_equipment_values(self, move_line):
+        if move_line.picking_id and move_line.picking_id.partner_id \
+                and move_line.picking_id.partner_id. \
+                service_location_id.id:
+            raise UserError(
+                _('Please select the Primary Service Location in %s') %
+                (move_line.picking_id.partner_id.name))
+        return {'name': '%s (%s)' % (move_line.product_id.name,
+                                     move_line.lot_id.name),
+                'product_id': move_line.product_id.id,
+                'lot_id': move_line.lot_id.id,
+                'current_location_id': move_line.picking_id and
+                move_line.picking_id.partner_id and
+                move_line.picking_id.partner_id and
+                move_line.picking_id.partner_id.
+                service_location_id and
+                move_line.picking_id.partner_id.
+                service_location_id.id or False}
```

Comment:
@murtuzasaleh The equipment should be created with the destination location. No need to check the primary location of the partner:

```
def prepare_equipment_values(self, move_line):
￼	        return {'name': '%s (%s)' % (move_line.product_id.name,
￼	                                     move_line.lot_id.name),
￼	                'product_id': move_line.product_id.id,
￼	                'lot_id': move_line.lot_id.id,
￼	                'current_location_id': move_line.dest_location_id}
```

Reply:
@max3903 This PR is WIP because as per mentioned in the wiki (Migrate to use stock_request instead of fsm.order.line and fsm.order.return) So I am working on it.